environs/context/callcontext.go

```diff
@@ -7,51 +7,67 @@ import (
 	stdcontext "context"
 
 	"github.com/juju/errors"
-	"github.com/juju/names/v4"
 
 	"github.com/juju/juju/domain/credential"
 )
 
+type (
+	// InvalidateModelCredentialFunc records a credential as being invalid.
+	InvalidateModelCredentialFunc func(reason string) error
+
+	// InvalidateModelCredentialFuncGetter returns a function which records a credential as being invalid.
+	InvalidateModelCredentialFuncGetter func() (InvalidateModelCredentialFunc, error)
+
+	// CredentialIDFuncGetter returns a function which returns a credential ID.
+	CredentialIDFuncGetter func() (credential.ID, error)
```

Comment:
This is not a function that returns a function. Please rename to `CredentialIDGetter`.

Reply:
Bah, it skewed, will fix.